sql/driver/driver_it_test.go

```diff
@@ -267,7 +270,11 @@ func TestSQLWithPortableData(t *testing.T) {
 			DoubleValue:   12.789,
```

Comment:
Here if I change rec and dec to be like this:

```go
                var dec types.Decimal
		rec := &Record{
			VarcharValue:  "hello",
			TinyIntValue:  -128,
			SmallIntValue: 32767,
			IntegerValue:  -27,
			BigIntValue:   38,
			BoolValue:     true,
			RealValue:     -5.32,
			DoubleValue:   12.789,
			DecimalValue:  dec,
		}

```

 and change Record struct to be:

```go

type Record struct {
	DecimalValue  types.Decimal
	NullValue     interface{}
	VarcharValue  string
	DoubleValue   float64
	BigIntValue   int64
	RealValue     float32
	IntegerValue  int32
	SmallIntValue int16
	TinyIntValue  int8
	BoolValue     bool
}
```


It fails with `runtime error: invalid memory address or nil pointer dereference` after named value checkers are run. Is this expected? Should not our type checker validate nil values?  The problem comes from the fact that dec is not nil, but its field unscaledValue is nil.

Reply:
We check `nil` values but `dec` is not a pointer, so it cannot be `nil`. The problem is, zero value of `types.Decimal` contains a `nil` `*big.Int`. I've updated the `*big.Int` serializer to handle that case by defaulting to `0` when a big int is `nil` at: https://github.com/hazelcast/hazelcast-go-client/pull/724/commits/3551354b914296e21d8d562eb03799936b631d47

I've also noticed some of the serializers were missing, added them along the way